qiskit/algorithms/time_evolvers/variational/variational_principles/variational_principle.py

```diff
@@ -0,0 +1,89 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class for a Variational Principle."""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from ....exceptions import AlgorithmError
+from ....gradients import BaseEstimatorGradient, BaseQGT, DerivativeType
+
+
+class VariationalPrinciple(ABC):
+    """A Variational Principle class. It determines the time propagation of parameters in a
+    quantum state provided as a parametrized quantum circuit (ansatz)."""
+
+    def __init__(
+        self,
+        qgt: BaseQGT | None = None,
+        gradient: BaseEstimatorGradient | None = None,
```

Comment:
Should None be supported here. While it is in derived classes they take care of None before calling super here. Since qgt and gradient are public instance vars any attempt to set them I think should not include None - assuming they need to be kept as public instance vars. In which case they also need docs.